### PR TITLE
chore: Prepare changelogs and version bump 1.2.0

### PR DESCRIPTION
**Pull request by miquelbeltran**
### Description :memo:

- **Purpose**: Changelog and version bump on `package.json`
- **Approach**: Update changelog and increase minor version.
- Depends on https://github.com/MindscapeHQ/raygun4node/pull/269
- Depends on https://github.com/MindscapeHQ/raygun4node/pull/263

**Type of change**

- [x] Chore

**Updates**

- Updated CHANGELOG
- Update `package.json`
- Update `package-lock.json` through the `npm install`

### Test plan :test_tube:

Run `npm publish --dry-run`

```
 npm publish --dry-run

> raygun@1.2.0 prepare
> tsc

npm notice
npm notice 📦  raygun@1.2.0
npm notice Tarball Contents
npm notice 6.4kB CHANGELOG.md
npm notice 21.4kB README.md
npm notice 0B build/.gitkeep
npm notice 1.3kB build/raygun.batch.d.ts
npm notice 5.5kB build/raygun.batch.js
npm notice 470B build/raygun.breadcrumbs.d.ts
npm notice 204B build/raygun.breadcrumbs.express.d.ts
npm notice 942B build/raygun.breadcrumbs.express.js
npm notice 4.2kB build/raygun.breadcrumbs.js
npm notice 4.4kB build/raygun.d.ts
npm notice 610B build/raygun.human.d.ts
npm notice 1.7kB build/raygun.human.js
npm notice 22.2kB build/raygun.js
npm notice 900B build/raygun.messageBuilder.d.ts
npm notice 9.1kB build/raygun.messageBuilder.js
npm notice 657B build/raygun.offline.d.ts
npm notice 3.7kB build/raygun.offline.js
npm notice 257B build/raygun.sync.transport.d.ts
npm notice 1.3kB build/raygun.sync.transport.js
npm notice 11B build/raygun.sync.worker.d.ts
npm notice 1.7kB build/raygun.sync.worker.js
npm notice 522B build/raygun.transport.d.ts
npm notice 2.4kB build/raygun.transport.js
npm notice 52B build/timer.d.ts
npm notice 408B build/timer.js
npm notice 4.9kB build/types.d.ts
npm notice 509B build/types.js
npm notice 2.1kB package.json
npm notice Tarball Details
npm notice name: raygun
npm notice version: 1.2.0
npm notice filename: raygun-1.2.0.tgz
npm notice package size: 26.5 kB
npm notice unpacked size: 97.9 kB
npm notice shasum: 30c0c2805121496c7d44ed051ff542ec0331e7f8
npm notice integrity: sha512-f09Y5IEqAuJoU[...]bKW6R7ZrPWhkw==
npm notice total files: 28
npm notice
npm notice Publishing to https://registry.npmjs.org/ with tag latest and public access (dry-run)
+ raygun@1.2.0
```

### Author to check :eyeglasses:
- [x] Project and all contained modules builds successfully
- [x] Self-/dev-tested 
- [x] Unit/UI/Automation/Integration tests provided where applicable
- [x] Code is written to standards
- [x] Appropriate documentation written (code comments, internal docs)

### Reviewer to check :heavy_check_mark:
- [x] Project and all contained modules builds successfully
- [ ] Change has been dev-/reviewer-tested, where possible
- [x] Unit/UI/Automation/Integration tests provided where applicable
- [x] Code is written to standards
- [x] Appropriate documentation written (code comments, internal docs)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,22 @@
+## 1.2.0
+
+TODO: Add the missing PRs here once merged!
+
+- feat: #65 Custom timestamp parameter (#254) (2024-06-19)
```
Comment (sumitramanga) [inline CHANGELOG.md:5]: Just wondering if the date is necessary if this all gets merged at once. Perhaps just one date?

**Comment by sumitramanga** [comment]
Everything else looks good and will ensure I check back once those 2 PR's are merged

File: CHANGELOG.md (hunk 1 above)
Comment (miquelbeltran) [reply CHANGELOG.md:5]: I will remove the dates.

**Comment by miquelbeltran** [comment]
Updated changelog with feature #218

**Review by sumitramanga — APPROVED** [review]
Looks good!! Yay for more improvements to the provider 🌟 Nice work

**Pull request merged**